core/src/main/java/com/scalar/db/storage/cassandra/CassandraAdmin.java

```diff
@@ -74,13 +98,29 @@ public void createNamespace(String namespace, Map<String, String> options)
       replicationOptions.put("class", ReplicationStrategy.NETWORK_TOPOLOGY_STRATEGY.toString());
       replicationOptions.put("dc1", replicationFactor);
     }
-    try {
-      clusterManager
-          .getSession()
-          .execute(query.with().replication(replicationOptions).getQueryString());
-    } catch (RuntimeException e) {
-      throw new ExecutionException(String.format("creating the keyspace %s failed", namespace), e);
-    }
+    String queryString = query.with().replication(replicationOptions).getQueryString();
+
+    clusterManager.getSession().execute(queryString);
+  }
+
+  private void createMetadataKeyspaceIfNotExists() {
+    CreateKeyspace query =
+        SchemaBuilder.createKeyspace(quoteIfNecessary(metadataKeyspace)).ifNotExists();
+    Map<String, Object> replicationOptions = new HashMap<>();
+    replicationOptions.put("class", ReplicationStrategy.SIMPLE_STRATEGY.toString());
+    replicationOptions.put("replication_factor", "1");
```

Comment:
If the replication factor should be configurable, should the replication strategy be too?

The way replication factor and strategy can be configured by the Schema Loader tool is very rough at the moment 
- only one strategy/replication factor can be set for all the tables in the schema file
- the network topology strategy cannot be configured properly. The datacenter names and a custom replication factor for each cannot be configured. (cf. the [code](https://github.com/scalar-labs/scalardb/blob/ea4c543465ca430dcbfa0eb627f8036779049351/core/src/main/java/com/scalar/db/storage/cassandra/CassandraAdmin.java#L70C1-L76))

So, if we want to configure the replication factor and strategy properly with the Schema Loader/Admin API, I think we need to rework this feature.

Reply:
I just remembered that we've already had this discussion.

> only one common strategy/replication factor can be set for all the keyspaces in the schema file

Actually, we have the option to configure the strategy/replication factor for each table as follows:
https://github.com/scalar-labs/scalardb/blob/08dd2117cc088a079a8b2197a686b17229757825/schema-loader/sample/schema_sample.json#L23-L24

But, in fact, the configuration is for a keyspace. Thus, being able to configure it for each table seems inconsistent.
And, as you mentioned, we can't properly configure the network topology strategy for now. So, I agree that we need to rework this feature in the future.

Given the current state, would it make sense to apply the same logic (using `options` to determine the replication factor and strategy) from `createKeyspace()` to the creation of the namespace meta table in `createMetadataKeyspaceIfNotExists()` with the same limitation?

Reply:
> Given the current state, would it make sense to apply the same logic (using options to determine the replication factor and strategy) from createKeyspace() to the creation of the namespace meta table in createMetadataKeyspaceIfNotExists() with the same limitation?

Yes, it does. I will do that in this PR. 

Concerning updating the default replication factor to 3, I will do it in a separate PR since I need to update multiple integration test classes to overwrite the replication factor back to 1.

Reply:
> Given the current state, would it make sense to apply the same logic (using options to determine the replication factor and strategy) from createKeyspace() to the creation of the namespace meta table in createMetadataKeyspaceIfNotExists() with the same limitation?

I revised it in https://github.com/scalar-labs/scalardb/pull/989/commits/fc2def1faecf44e9de2ea88c07560db7693cb92a